Store Run Status

JamesVaughan opened the pull request:
Addresses issue #35.
Implements the storage of error messages or a completion status into a model system run's directory.

Implements the Validation check when constructing the model system.
Implements the RuntimeValidation check for the model system.

JamesVaughan committed:
Implements the storage of error messages or a completion status into a model system run's directory.
Implements the Validation check when constructing the model system.
Implements the RuntimeValidation check for the model system.

BrendanReilly commented on src/XTMF2/RunResults.cs:
```diff
@@ -0,0 +1,192 @@
+        /// </summary>
+        /// <param name="runDirectory">The directory that the run was completed in.</param>
+        /// <returns></returns>
+        internal static void WriteRunCompelted(string runDirectory)
```
Minor, just spelling on completed

BrendanReilly commented on tests/XTMF2.UnitTests/Modules/FailsAtRuntime.cs:
```diff
@@ -0,0 +1,35 @@
+using System.Text;
+using XTMF2;
+
+namespace TestXTMF.Modules
```
Is this namespace better xtmf2.testing ? Or TestXTMF2

BrendanReilly commented on tests/XTMF2.UnitTests/TestRun.cs:
```diff
@@ -216,32 +214,225 @@ public void ReportRunProgress()
+                        success = true;
+                        sim.Release();
+                    };
+                    runBus.ClientErrorWhenRunningModelSystem += (sender, runId, e, stack) =>
```
Does it make sense to use OnErrorXYZ language to show it's an event?

JamesVaughan replied:
Following Microsoft style for WPF it looks like events do not get the OnXYZ pattern. On is included for methods for those events that can be overridden.

BrendanReilly reviewed (approved):
Approved, comments not necessary changes

JamesVaughan committed:
Fixed the namespace for test modules to XTMF2.Modules.

Fixed the Ignore modules to be properly added to the namespace XTMF2.RuntimeModules
